src/routes/events.js

```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+const {
+    EVENT_VALIDATION_RULES,
+    handleValidation,
+} = require('../utility/validation');
+
+const router = express.Router();
+const eventController = require('../controllers/events');
+
+router.post(
+    '/:id/join',
```

Comment:
Join is a verb. Endpoints should not contain verbs. Now you are saying something "Post user as join." Better is to say "Post user as volunteer. 
So, I think this should be changed to volunteer.